### ISSUE TITLE
Add doc for configurations.

### ISSUE THREAD

**Issue by liurenjie1024** [risingwavelabs/risingwave-docs-legacy#997]
Currently, we don't have documentation for our configurations. As an example, flink has excellent support for this: https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/deployment/config/

We also need to set up similar page so that our customer can understand and tune risingwave by themselves.

**Comment by neverchanje** [comment]
I think that we should only list the key configurations that we want developers to tune instead of documenting all configurations, which will be hard to maintain.

So, would you please help list the key configs first? @liurenjie1024

**Comment by neverchanje** [comment]
As we discussed offline, we can use the rust doc as the reference for the configurations.


https://risingwavelabs.github.io/risingwave/risingwave_common/config/struct.RwConfig.html

**Comment by hengm3467** [comment]
I think we need:
1) the list of configurable parameters.
2) How to configure these parameters (best practices). For example, `streaming_parallelism` can be 2 or 100, how do we determine the specific number. 

If you or someone else can help with summarizing the parameters and the best practices, I can put them into our docs. @liurenjie1024 @neverchanje

**Comment by hengm3467** [comment]
@liurenjie1024 & @neverchanje Since we already have documentation for all the runtime parameters and system parameters, we can keep adding new notes and tips to these pages:

Runtime parameters:
https://www.risingwave.dev/docs/current/view-configure-runtime-parameters/

System parameters:
https://www.risingwave.dev/docs/current/view-configure-system-parameters/

**Comment by hengm3467** [comment]
Will cover the details of parameter settings in a separate document set "Optimize performance".

**Comment by liurenjie1024** [comment]
Hi, @hengm3467 I took a look at the links and it's different from what I mean. I'll have a talk with you later to figure out the list of prarmeters, but for now let's keep it open.

**Comment by neverchanje** [comment]
We need especially a doc about how to update the configurations online. For more specific configurations, I think we don't document them all for now.  We only need to cover a subset.

**Comment by hengm3467** [comment]
@neverchanje Did you mean all the options in risectl?

**Comment by neverchanje** [comment]
No. I mean the configs in this place: https://github.com/risingwavelabs/risingwave/blob/main/src/config/example.toml Although most of the configurations are not user-facing, some of them are critical to system performance.

Eventually, we may need a page like this: https://www.mongodb.com/docs/manual/reference/configuration-options/

In order to maintain the document with minimal efforts, I need to write an automation tool that extracts comments in the code, which will be used as config description.

**Comment by neverchanje** [comment]
I recently developed a tool that can generates configs from comments in the code. The result is a markdown file, which can be directly used without modication in the doc. https://github.com/risingwavelabs/risingwave/pull/14727
